### PR TITLE
Typed: Factory methods for dispatcher selector props 

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22828

### CODE REVIEW

File: akka-typed/src/main/scala/akka/typed/Props.scala [hunk 1]
```diff
@@ -164,17 +164,54 @@ private[akka] case object EmptyProps extends Props {
...
- * INTERNAL API
  */
-@DoNotInherit
-@InternalApi
-private[akka] sealed trait DispatcherSelector extends Props
+object DispatcherSelector {
+
+  /**
+   * Scala API:
+   * Run the actor on the same executor as its parent.
```
Comment (patriknw) [inline akka-typed/src/main/scala/akka/typed/Props.scala]: ah, is that what it is for Akka Typed (not the `default-dispatcher`)?

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-typed/src/main/scala/akka/typed/Props.scala (hunk 1 above)
Comment (johanandren) [inline akka-typed/src/main/scala/akka/typed/Props.scala]: That's what the docs of the class said, I'd have to check to say for sure. But I think it makes sense (and has been something users ask every now and then)

Comment (patriknw) [inline akka-typed/src/main/scala/akka/typed/Props.scala]: yes, it's a better default

**Pull request merged**